PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1

```diff
@@ -266,177 +266,181 @@ function Get-PrivilegedUser {
-                                $PrivilegedUsers[$GroupMember.Id].roles += $Role.DisplayName
-                            }
-                        }
-                    }
-
-                    # Get the users that are assigned to the PIM group as Eligible members
-                    $graphArgs = @{
-                        "commandlet" = "Get-MgBetaIdentityGovernancePrivilegedAccessGroupEligibilityScheduleInstance"
-                        "queryParams" = @{'$filter' = "groupId eq '$UserObjectId'"}
-                        "M365Environment" = $M365Environment}
-                    $PIMGroupMembers = Invoke-GraphDirectly @graphArgs
-                    foreach ($GroupMember in $PIMGroupMembers) {
-                        # If the user is not a member of the PIM group (i.e. they are an owner) then skip them
-                        if ($GroupMember.AccessId -ne "member") { continue }
-                        $PIMEligibleUserId = $GroupMember.PrincipalId
-
-                        # If the user's data has not been fetched from graph, go get it
-                        if (-not $PrivilegedUsers.ContainsKey($PIMEligibleUserId)) {
-                            $AADUser = Get-MgBetaUser -ErrorAction Stop -UserId $PIMEligibleUserId
-                            $PrivilegedUsers[$PIMEligibleUserId] = @{"DisplayName"=$AADUser.DisplayName; "OnPremisesImmutableId"=$AADUser.OnPremisesImmutableId; "roles"=@()}
-                        }
-                        # If the current role has not already been added to the user's roles array then add the role
-                        if ($PrivilegedUsers[$PIMEligibleUserId].roles -notcontains $Role.DisplayName) {
-                            $PrivilegedUsers[$PIMEligibleUserId].roles += $Role.DisplayName
-                        }
-                    }
-                }
+        # Since this is a group, we need to also process assignments in PIM
+        # If the premium license for PIM is there, process the users that are "member" of the PIM group as Eligible
+        if ($TenantHasPremiumLicense) {
+            # Get the users that are assigned to the PIM group as Eligible members
+            $graphArgs = @{
+                "commandlet" = "Get-MgBetaIdentityGovernancePrivilegedAccessGroupEligibilityScheduleInstance"
+                "queryParams" = @{'$filter' = "groupId eq '$GroupId'"}
+                "M365Environment" = $M365Environment }
+            $PIMGroupMembers = Invoke-GraphDirectly @graphArgs
+            foreach ($GroupMember in $PIMGroupMembers) {
+Write-Warning "Processing role: $($Role.DisplayName) PIM group Eligible member: $($GroupMember.PrincipalId)"
+                # If the user is not a member of the PIM group (i.e. they are an owner) then skip them
+                if ($GroupMember.AccessId -ne "member") { continue }
+                $PIMEligibleUserId = $GroupMember.PrincipalId
+
+                $Recursioncount++
```

Comment:
can PIM groups be added to PIM groups?  Or is it just 

PIM group contains group containers users?

How deep can this potentially go?

Reply:
It can go at least 3-4 levels deep according to my testing but I suspect it can go much deeper than that and I am not sure if there is a limit. There is also a plethora of ways that groups can be nested. I documented my testing on this subject in #1303 for reference.

Reply:
Ok.  I think having the recursion limit you have is fine for now.  Thanks for the research.